backend/src/db/migrations/20200326160326-remove_dangling_image_urls.js

```diff
@@ -0,0 +1,61 @@
+import { getDriver } from '../../db/neo4j'
+import { existsSync } from 'fs'
+
+export const description = `
+  In this review:
+  https://github.com/Human-Connection/Human-Connection/pull/3262#discussion_r398634249
+  I brought up that we may have image nodes with danling urls (urls that don't
+  point to local file on disk). I would prefer to remove those urls to avoid
+  unnecessary 404 errors.
+`
+
+export async function up(next) {
+  const driver = getDriver()
+  const session = driver.session()
+  const transaction = session.beginTransaction()
+
+  try {
+    // Implement your migration here.
+    const { records } = await transaction.run(`
+      MATCH(image:Image)
+      WHERE image.url STARTS WITH '/'
+      RETURN image.url as url
+    `)
+    const urls = records.map((record) => record.get('url'))
+    const danglingUrls = urls.filter((url) => {
+      const fileLocation = `public${url}`
+      return !existsSync(fileLocation)
+    })
+    await transaction.run(
+      `
+      MATCH(image:Image)
+      WHERE image.url IN $danglingUrls
+      DETACH DELETE image
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-26T16:16:11Z" title="Thursday, March 26th 2020, 5:16:11 pm +01:00">Mar 26, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,61 @@
+import { getDriver } from '../../db/neo4j'
+import { existsSync } from 'fs'
+
+export const description = `
+  In this review:
+  https://github.com/Human-Connection/Human-Connection/pull/3262#discussion_r398634249
+  I brought up that we may have image nodes with danling urls (urls that don't
+  point to local file on disk). I would prefer to remove those urls to avoid
+  unnecessary 404 errors.
+`
+
+export async function up(next) {
+  const driver = getDriver()
+  const session = driver.session()
+  const transaction = session.beginTransaction()
+
+  try {
+    // Implement your migration here.
+    const { records } = await transaction.run(`
+      MATCH(image:Image)
+      WHERE image.url STARTS WITH '/'
+      RETURN image.url as url
+    `)
+    const urls = records.map((record) => record.get('url'))
+    const danglingUrls = urls.filter((url) => {
+      const fileLocation = `public${url}`
+      return !existsSync(fileLocation)
+    })
+    await transaction.run(
+      `
+      MATCH(image:Image)
+      WHERE image.url IN $danglingUrls
+      REMOVE image.url
```


I don't think it's enough to `REMOVE image.url`
I believe we need to `DETACH DELETE image`, oder?